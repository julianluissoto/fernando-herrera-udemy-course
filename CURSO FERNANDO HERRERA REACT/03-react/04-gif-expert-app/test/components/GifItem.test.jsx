import { render, screen } from "@testing-library/react";
import GifItem from "../../src/components/GifItem";

describe(" TEST GifItem component", () => {
  const title = "Saitama";
  const url = "http://one-punch.com/saitama.png";

  test("should match with the snapshot ", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("must show the image with the url and indicated ALT", () => {
    render(<GifItem title={title} url={url} />);
    //screen.debug();

    //expect(screen.getByRole("img").src).toBe(url);  way 1
    //expect(screen.getByRole("img").alt).toBe(title); way  1

    //with destructuring

    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });
  test("the title must show on p tag", () => {
    render(<GifItem title={title} url={url} />);

    expect(screen.getByText(title)).toBeTruthy();
  });
});
