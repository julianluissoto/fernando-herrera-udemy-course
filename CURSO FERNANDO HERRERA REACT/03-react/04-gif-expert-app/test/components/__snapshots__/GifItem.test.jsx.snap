// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` TEST GifItem component should match with the snapshot  1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="http://one-punch.com/saitama.png"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
